inst/include/Rcpp/sugar/functions/rowSums.h

```diff
@@ -0,0 +1,963 @@
+        for (j = 0; j < nc; j++) {                                                                              \
+            if (n_ok[j]) {                                                                                      \
+                detail::div(res[j], n_ok[j]);                                                                   \
+            } else {                                                                                            \
+                detail::set_nan(res[j]);                                                                        \
+            }                                                                                                   \
+        }                                                                                                       \
+                                                                                                                \
+        return res;                                                                                             \
+    }                                                                                                           \
+};
+
+COL_MEANS_IMPL_RMNA(LGLSXP) 
+COL_MEANS_IMPL_RMNA(INTSXP) 
+
+
+//  ColMeans
+//      Input with template parameter NA = false
+//      ColMeansImpl<..., NA_RM = false>
+//
+template <int RTYPE, typename T, bool NA_RM>
+class ColMeansImpl<RTYPE, false, T, NA_RM>
+    : public ColMeansImpl<RTYPE, false, T, false> {};
+
+
+} // sugar
+
+
+template <int RTYPE, bool NA, typename T>
+inline typename sugar::detail::RowSumsReturn<RTYPE>::type
+rowSums(const MatrixBase<RTYPE, NA, T>& x, bool na_rm = false) {
+    if (!na_rm) {
+        return sugar::RowSumsImpl<RTYPE, NA, T, false>(x);
+    }
+    return sugar::RowSumsImpl<RTYPE, NA, T, true>(x);
+}
+
+template <int RTYPE, bool NA, typename T>
+inline typename sugar::detail::ColSumsReturn<RTYPE>::type
+colSums(const MatrixBase<RTYPE, NA, T>& x, bool na_rm = false) {
+    if (!na_rm) {
+        return sugar::ColSumsImpl<RTYPE, NA, T, false>(x);
+    }
+    return sugar::ColSumsImpl<RTYPE, NA, T, true>(x);
+}
+
+template <int RTYPE, bool NA, typename T>
+inline typename sugar::detail::RowMeansReturn<RTYPE>::type
+rowMeans(const MatrixBase<RTYPE, NA, T>& x, bool na_rm = false) {
+    if (!na_rm) {
+        return sugar::RowMeansImpl<RTYPE, NA, T, false>(x);
+    }
+    return sugar::RowMeansImpl<RTYPE, NA, T, true>(x);
+}
+
+template <int RTYPE, bool NA, typename T>
+inline typename sugar::detail::ColMeansReturn<RTYPE>::type
+colMeans(const MatrixBase<RTYPE, NA, T>& x, bool na_rm = false) {
+    if (!na_rm) {
+        return sugar::ColMeansImpl<RTYPE, NA, T, false>(x);
+    }
+    return sugar::ColMeansImpl<RTYPE, NA, T, true>(x);
+}
+
+
+#undef ROW_SUMS_IMPL_KEEPNA
+#undef ROW_SUMS_IMPL_RMNA
+#undef COL_SUMS_IMPL_KEEPNA
+#undef COL_SUMS_IMPL_RMNA
+#undef ROW_MEANS_IMPL_KEEPNA
+#undef ROW_MEANS_IMPL_RMNA
+#undef COL_MEANS_IMPL_KEEPNA
+#undef COL_MEANS_IMPL_RMNA
```

Comment:
Oh, I see now that everything is `#undef`ed at the bottom here. Marginal preference to `#undef` immediately after you know you don't need it anymore, but I think this is fine too.